app/src/main/kotlin/io/homeassistant/companion/android/settings/SettingsActivity.kt

```diff
@@ -42,8 +43,16 @@ class SettingsActivity : BaseActivity() {
     private var externalAuthCallback: ((Int) -> Boolean)? = null
 
     companion object {
-        fun newInstance(context: Context): Intent {
-            return Intent(context, SettingsActivity::class.java)
+        const val SCREEN_DEVELOPER = "developer"
+        const val SCREEN_NOTIFICATION_HISTORY = "notification_history"
+        const val SCREEN_WEBSOCKET = "websocket"
+
+        fun newInstance(context: Context, screen: String? = null): Intent {
```

Comment:
Make an enum out of the potential screens.

Reply:
Adjusted, even though this significantly increases the number of changes not strictly related to the feature.

Reply:
Adjusted, even though this significantly increases the number of changes not strictly related to the feature.

@TimoPtr Could you check if this looks good to you? (if so and no new comments it can be merged, otherwise I'll address them on Saturday)

Reply:
All good thanks 👍🏻. In the end we could have use a sealed class to handle the parameters of the deep link but it's good enough to expose the API of the fragment.